tchannel-core/src/main/java/com/uber/tchannel/codecs/CodecUtils.java

```diff
@@ -89,7 +89,7 @@ public static void encodeSmallString(String value, ByteBuf buffer) {
     public static Map<String, String> decodeHeaders(ByteBuf buffer) {
 
         int numHeaders = buffer.readUnsignedShort();
-        Map<String, String> headers = new HashMap<String, String>(numHeaders);
+        Map<String, String> headers = new HashMap<>(numHeaders);
```

Comment:
this will cause a guaranteed re-sizing of the map. use `com.google.common.collect.Maps.newHashMapWithExpectedSize(numHeaders)` instead.